ratpack-core/src/main/java/ratpack/http/internal/ByteBufBackedTypedData.java

```diff
@@ -67,7 +67,10 @@ public String getText(Charset charset) {
   @Override
   public byte[] getBytes() {
     if (byteBuf.hasArray()) {
-      return byteBuf.array();
+      byte[] bytes = new byte[byteBuf.readableBytes()];
+      int readerIndex = byteBuf.readerIndex();
+      byteBuf.getBytes(readerIndex, bytes);
```

Comment:
This will be a copy in this case does it make sense to make a new method? We could also make clear that this is the raw array which may be whats wanted in some cases?

Reply:
It does make sense to make it more plain what the method actually does, either in the Javadoc or by changing the method name. And I do see the value of having a copy method because it's easy to mess up the copying if some external code only works with byte arrays. Changing the method name could break some existing code, of course that code had probably worked correctly only by coincidence :)

The current behaviour of returning the raw array itself is probably a bit pointless though because you also need `readerIndex` and `readableBytes` to use it correctly so you might as well work with the `ByteBuf` directly then.

Please advise, I can modify the patch either way.

Reply:
OK - should I mark the old method as `@deprecated` or something? I don't know, I just don't see the reason for it existing as it does now (the other branch of the `if` statement will _still_ make an array copy, so users should either use the `ByteBuf` or explicitly call the new `copyBytes`)?